## Plausible Lengths
My assumption of plausible lengths for bridges in Pittsburgh are between 1,000 and 2,000ft. 

In [1]:
import pandas as pd
df = pd.read_csv("bridges.csv")
display(df.shape)
display(df.head(10))
display(df.describe())

(108, 13)

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
0,E1,M,3.0,1818,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
1,E2,A,25.0,1819,HIGHWAY,1037.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
2,E3,A,39.0,1829,AQUEDUCT,NaN,1.0,N,THROUGH,WOOD,NaN,S,WOOD
3,E5,A,29.0,1837,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,SHORT,S,WOOD
4,E6,M,23.0,1838,HIGHWAY,NaN,2.0,N,THROUGH,WOOD,NaN,S,WOOD
5,E7,A,27.0,1840,HIGHWAY,990.0,2.0,N,THROUGH,WOOD,MEDIUM,S,WOOD
6,E8,A,28.0,1844,AQUEDUCT,1000.0,1.0,N,THROUGH,IRON,SHORT,S,SUSPEN
7,E9,M,3.0,1846,HIGHWAY,1500.0,2.0,N,THROUGH,IRON,SHORT,S,SUSPEN
8,E10,A,39.0,1848,AQUEDUCT,NaN,1.0,N,DECK,WOOD,NaN,S,WOOD
9,E11,A,29.0,1851,HIGHWAY,1000.0,2.0,N,THROUGH,WOOD,MEDIUM,S,WOOD


,location,erected,length,lanes
count,107.000000,108.000000,81.000000,92.000000
mean,25.978505,1905.314815,1567.469136,2.630435
std,13.665871,37.173664,747.491523,1.164657
min,1.000000,1818.000000,804.000000,1.000000
25%,15.500000,1884.000000,1000.000000,2.000000
50%,27.000000,1903.000000,1300.000000,2.000000
75%,37.500000,1928.000000,2000.000000,4.000000
max,52.000000,1986.000000,4558.000000,6.000000


In [3]:
df.isna().sum()

Id           0
river        0
location     1
erected      0
purpose      0
length      27
lanes       16
clear-g      2
t-or-d       6
material     2
span        16
rel-l        5
type         2
dtype: int64

## Findings

From my findings from the data given it appears as if the minimum length recorded is 804 feet, while the maximum is 4558 feet. While the minimum is pretty close to my initial assumption, the maximum is more than double in length what my initial assumption was.

Because of this I took to conducting a web search on Pittsburgh bridge lengths and found that the shortest bridge is actually 105 feet while the longest bridge is almost 7300 feet. Within these parameters, all of the data that we have available is reasonable. Though it also shows that it may be incomplete. 

In our dataset, we are missing 27 lengths. While normally if we were to impute we would estimate or average these numbers. But since bridge lengths are such exact numbers, I don't believe that this would be beneficial to us. As shown from my research there are bridges that are approx 700 feet shorter than the shortest one recorded and approx 2800 feet longer than the longest recorded. 

With this information in mind, I would delete the rows that are missing the length. If straight up deletion is not an option, I would impute the rows conditionally; based on the span column. And if that column is missing, I would again delete that row. 

https://github.com/skeown13/cs82a-data-science-portfolio